### PR TITLE
add check for remote Repository in GitDriver::supports

### PR DESCRIPTION
**Pull request by bragento**

### CODE REVIEW

**Comment by GromNaN** [comment]
For repositories with a lot of tags / branches, this can takes a lot of time (21s on for `symfony/symfony` on my server)

```
$ git ls-remote https://github.com/symfony/symfony.git | wc -l
15566
```

This command can be a little faster:
```
git ls-remote --heads https://github.com/symfony/symfony.git
```

File: src/Composer/Repository/Vcs/GitDriver.php [hunk 1]
```diff
@@ -241,7 +241,11 @@ public static function supports(IOInterface $io, Config $config, $url, $deep = f
             return false;
         }
 
-        // TODO try to connect to the server
+        $process = new ProcessExecutor($io);
+        if($process->execute('git ls-remote ' . $url, $output) === 0) {
```
Comment (stof) [inline src/Composer/Repository/Vcs/GitDriver.php]: you need to escape the URL.

**Comment by bragento** [comment]
added --heads param and escaped the repo url

**Comment by GromNaN** [comment]
:+1:

**Comment by Seldaek** [comment]
Thanks!

**Pull request merged**